pkg/kv/kvpb/api.proto

```diff
@@ -3689,6 +3689,79 @@ message JoinNodeResponse {
   roachpb.Version active_version = 4;
 }
 
+// DistSender defines a subset of the Internal service RPCs, specifically for
+// DRPC usage within the DistSender subsystem.
+//
+// During the gRPC to DRPC migration:
+// - gRPC clients will continue to use the full 'Internal' service client.
+// - DRPC clients targeting DistSender functionalities will use clients
+//   generated from this 'DistSender' service.
+// - The 'RestrictedInternalClient' interface ensures that components like
+//   DistSender can operate with either the gRPC 'Internal' client or this new
+//   DRPC 'DistSender' client.
+//
+// TODO: Remove these methods from the 'Internal' service once the DRPC
+// migration for DistSender is complete, making this service the sole
+// definition.
+service DistSender {
```

Comment:
Is `DistSender` the correct name? This service lives on the `Node` and represents the internal KV endpoints. `DistSender` is who is primarily talking to those endpoints. I thought (assumed, really) that the service name would reflect the receiver, rather than the dominant sender - in which case `KV` may be a better name (also nice and short).

Have you considered splitting MuxRangeFeed out as well? This could aid with the migration, since they're fairly different RPCs. The service could be named (surprise) `Rangefeed`.